pkg/cleanup/cmd/cleanup/cleanup.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2024 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"strings"
+
+	"go.uber.org/zap"
+	v1 "k8s.io/api/apps/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"knative.dev/pkg/environment"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+)
+
+func main() {
+	logger := setupLogger()
+	defer logger.Sync()
+
+	env := environment.ClientConfig{}
+	env.InitFlags(flag.CommandLine)
+
+	flag.Parse()
+
+	config, err := env.GetRESTConfig()
+	if err != nil {
+		logger.Fatalf("failed to get kubeconfig %s", err)
+	}
+
+	client := kubernetes.NewForConfigOrDie(config)
+
+	logger.Info("Deleting old Serving resources if any")
+
+	// Delete all deployments
+	var deps *v1.DeploymentList
+	if deps, err = client.AppsV1().Deployments(system.Namespace()).List(context.Background(), metav1.ListOptions{}); err != nil {
+		logger.Fatal("failed to list deployments: ", err)
+	}
+	for _, dep := range deps.Items {
+		if hasPrefix(dep.Name) {
```

Comment:
Don't we know the exact names of the deployments we want to delete?

Reply:
Probably a relic need to update.